plugins/inputs/logfile/tail/tail_queue.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tail
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/fdlimit"
+)
+
+/*
+	Centralize all the tailers which are scheduled for each file
+*/
+
+var defaultOpenFileLimit int = 1024
+
+type TailerEnqueue struct {
+	Queue chan bool
```

Comment:
I strongly disagree with this. Why are we storing a boolean here? Do we need to check the value? I thought this was just a counter. 

Ultimately what you are implementing is a semaphore with a buffered channel. If all we want is a counting semaphore, we could just do `chan struct{}`. Otherwise we're exploding memory usage.

Reply:
Or instead of using a channel, we could just use `atomic` operations on a global count. But I would probably just use a chan struct{}

Reply:
> Why are we storing a boolean here?

In future, TailerEnqueue could  change to store `tailer` instead of `struct`. Therefore, it would be best to prepare the behavior changes  for the customer `now`. However, if we only limit the scope to only log number of `monitored files`, then I agree with you regarding using `struct`.

Reply:
Did we discuss storing a duplicate tailer object in a channel before? I disagree. The entire purpose of this is counting open monitored files. Why do we need more information than a count?

Reply:
👍  Yes, maybe I'm worrying about too much about things out of scope. Will change back in the latest commit.